main_summarizer.py

```diff
@@ -0,0 +1,21 @@
+from transformers import pipeline
+
+
+class Summarizer:
+	MAX_TOKEN_COUNT = 1024
+	TEXT_LENGTH_INTERVAL = 100
+
+	def __init__(self):
+		self.pipe = pipeline("summarization", model="facebook/bart-large-cnn")
+
+	def summarize(self, text, min_size):
+		trimmed_text = text
+
+		while len(self.pipe.tokenizer(trimmed_text).tokens()) > self.MAX_TOKEN_COUNT:
```

Comment:
Tokenization is a deterministic process, you should call it once.

Reply:
But I'm changing the variable trimmed_text in a loop, what do you mean by "deterministic" when the function is called with different inputs?

Reply:
You can do something like that:
```python
tokens = tokenizer(text).tokens()
tokens = tokens[:self.MAX_TOKEN_COUNT]
```